### PR TITLE
Code2class

### PR DESCRIPTION
**Pull request by maximzubkov**
Add `code2class` model and refactor `DataModules`

### CODE REVIEW

**Comment by maximzubkov** [comment]
In both PR tests are not passed due to the problems in `torchvision`, the problem is discussed in the [issue](https://github.com/pytorch/vision/issues/3554)

**Comment by maximzubkov** [comment]
I decided to switch from the `MNIST` dataset to `CIFAR10`

File: dataset/data_modules/text_data_module.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    ):
+        super().__init__(
+            dataset_name=dataset_name,
+            batch_size=batch_size,
+            is_test=is_test,
+            transform=transform,
+            num_classes=num_classes
+        )
+
+    @abstractmethod
```
Comment (SpirinEgor) [inline dataset/data_modules/text_data_module.py:28]: It is strange that you mark a method as abstract and then immediately implement it. Maybe it's better to remove this decorator?

**Review by SpirinEgor — CHANGES_REQUESTED** [review]
Have few questions

**Review by SpirinEgor — APPROVED** [review]

**Pull request merged**